### PR TITLE
revise 'Examples' of LBFGS to create right docs(cn) and add a new method for right docs(en)

### PR DESCRIPTION
**Pull request by lijialin03**
### PR types

Others
### PR changes

Docs
### Description

Pcard-66961
revise 'Examples' of public functions of LBFGS to create right docs(cn) and add a new method for right docs(en)

### CODE REVIEW

File: python/paddle/optimizer/lbfgs.py [hunk 1]
```diff
@@ -515,6 +553,38 @@ def step(self, closure):
         Args:
             closure (callable): A closure that reevaluates the model
                 and returns the loss.
```
Comment (sunzhongkai588) [inline python/paddle/optimizer/lbfgs.py]: 由于rst格式要求，该行开头`and`要和上一行开头`closure`对齐，否则无法解析
image

File: python/paddle/optimizer/lbfgs.py [hunk 2]
```diff
@@ -515,6 +553,38 @@ def step(self, closure):
         Args:
```
Comment (sunzhongkai588) [inline python/paddle/optimizer/lbfgs.py:554]: `Args: `上方需要空一行

File: python/paddle/optimizer/lbfgs.py [hunk 3]
```diff
@@ -454,6 +453,45 @@ def state_dict(self):
         Return:
             state, a dict holding current optimization state. Its content
                 differs between optimizer classes.
```
Comment (sunzhongkai588) [inline python/paddle/optimizer/lbfgs.py]: 由于rst格式要求，该行开头`differs`要和上一行开头`state`对齐，否则解析错误

image

**Review by sunzhongkai588 — DISMISSED** [review]
辛苦，LGTM～

File: python/paddle/optimizer/lbfgs.py [hunk 4]
```diff
@@ -453,7 +452,46 @@ def state_dict(self):
...
+                        loss.backward()
+                        return loss
+
+                    opt.step(closure)
+
+                inputs = paddle.rand([10, 10], dtype="float32")
+                targets = paddle.to_tensor([2 * x for x in inputs])
+
+                n_iter = 0
+                while n_iter < 20:
```
Comment (cxxly) [inline python/paddle/optimizer/lbfgs.py:490]: for...range... is more pythonic

File: python/paddle/optimizer/lbfgs.py [hunk 5]
```diff
@@ -699,3 +770,11 @@ def obj_func(x, alpha, d):
...
 
         return orig_loss
+
+    def minimize(
+        self, loss, startup_program=None, parameters=None, no_grad_set=None
+    ):
+        """Empty method. LBFGS optimizer does not use this way to minimize ``loss``."""
+        raise NotImplementedError(
+            "LBFGS optimizer does not use this way to minimize loss."
+        )
```
Comment (cxxly) [inline python/paddle/optimizer/lbfgs.py:780]: please refer xxx usage....

**Review by cxxly — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by sunzhongkai588 — APPROVED** [review]
LGTM for docs

**Pull request merged**